web/frontend/src/components/slam.vue

```diff
@@ -39,8 +41,10 @@ const concatArrayU8 = (arrays: Uint8Array[]) => {
   return result;
 };
 
-const fetchSLAMMap = (name: string) => {
-  return nextTick(() => {
+const fetchSLAMMap = (name: string): Promise<Uint8Array | undefined> => {
+  return new Promise((resolve, reject) => {
```

Comment:
I'm not sure why this needs to return a Promise instead of just blocking until it returns the map, since we always call await on that Promise immediately.